### PR TITLE
Rust v1.63.0

### PR DESCRIPTION
**Pull request by brennanwatt**
Updating rust version. See #25489

### CODE REVIEW

File: ci/test-checks.sh [hunk 1]
```diff
@@ -65,11 +65,25 @@ fi
...
+  # This lint occurs all over the code base
+  "--allow=clippy::significant_drop_in_scrutinee"
+
+  # The prost crate, used by solana-storage-proto, generates Rust source that
+  # triggers this lint. Need to resolve upstream in prost
+  "--allow=clippy::derive_partial_eq_without_eq"
+
+  # This link seems to incorrectly trigger in
+  # `programs/bpf_loader/src/syscalls/{lib,cpi}.rs`
+  "--allow=clippy::explicit-auto-deref"
```
Comment (joncinque) [inline ci/test-checks.sh:78]: nittiest of nits: for consistency, let's have this be 
```suggestion
  "--allow=clippy::explicit_auto_deref"
```

File: runtime/src/in_mem_accounts_index.rs [hunk 2]
```diff
@@ -1416,6 +1416,7 @@ impl<'a> FlushGuard<'a> {
     /// Set the `flushing` atomic flag to true.  If the flag was already true, then return `None`
     /// (so as to not clear the flag erroneously).  Otherwise return `Some(FlushGuard)`.
     #[must_use = "if unused, the `flushing` flag will immediately clear"]
+    #[allow(clippy::unnecessary_lazy_evaluations)]
```
Comment (joncinque) [inline runtime/src/in_mem_accounts_index.rs:1419]: Someone will probably inadvertently change this at some point -- any chance we can add the `allow` specifically on the offending line and / or a comment explaining why this is needed?

**Review by joncinque — COMMENTED** [review]
R+ after the two small things I brought up

File: runtime/src/in_mem_accounts_index.rs (hunk 2 above)
Comment (brennanwatt) [reply runtime/src/in_mem_accounts_index.rs]: Done

File: ci/test-checks.sh (hunk 1 above)
Comment (brennanwatt) [reply ci/test-checks.sh]: Done

**Review by joncinque — APPROVED** [review]

**Comment by brennanwatt** [comment]
Kicking off some validator testing to make sure nothing pops up. @mvines - Let me know if there is anything else that needs to be done or reviewed before merging

**Comment by mvines** [comment]
> Kicking off some validator testing to make sure nothing pops up. @mvines - Let me know if there is anything else that needs to be done or reviewed before merging

lgtm! I saw this review already two approvals so no need for me to "me2!" it.

Thanks for doing this work btw

**Pull request merged**